policy-controller/k8s/index/src/outbound/index/unmeshed_network.rs

```diff
@@ -0,0 +1,230 @@
+use chrono::{offset::Utc, DateTime};
+use linkerd_policy_controller_k8s_api::policy::Cidr;
+use linkerd_policy_controller_k8s_api::{policy as linkerd_k8s_api, ResourceExt};
+use std::net::IpAddr;
+
+#[derive(Debug, Default)]
+pub(crate) struct Networks(pub Vec<Cidr>);
+
+#[derive(Debug, Default)]
+pub(crate) struct UnmeshedNetwork {
+    pub networks: Networks,
+    pub name: String,
+    pub namespace: String,
+    pub creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedUnmeshedNetwork {
+    matched_cidr: MatchedCidr,
+    name: String,
+    namespace: String,
+    creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedCidr(Cidr);
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl Networks {
+    pub fn find_match(&self, ip: IpAddr) -> Option<Cidr> {
+        let ip: Cidr = ip.into();
+        self.0
+            .iter()
+            .filter(|c| c.contains(&ip))
+            .min_by(|a, b| a.size().cmp(&b.size()))
+            .cloned()
+    }
```

Comment:
this is only used in one place, here inside of this submodule. does it need to be public?

zooming out more, do we get much benefit from defining a `Networks` newtype here? `Networks` only provides this interface, called once lower down `resolve_unmeshed_network`.

Reply:
Yeah, you are right. We really do not. Getting rid of it 🎉